src/v/pandaproxy/rest/handlers.cc

```diff
@@ -81,29 +83,40 @@ get_brokers(server::request_t rq, server::reply_t rp) {
     auto make_metadata_req = []() {
         return kafka::metadata_request{.list_all_topics = false};
     };
-    return rq.service()
-      .client()
-      .local()
-      .dispatch(make_metadata_req)
-      .then([res_fmt, rp = std::move(rp)](
+    // Servicing the request must be handled on the
+    // same core where the client is held. So we capture
+    // everything in a handler and pass that to the cache.
+    credential_t user{rq.ctx.user};
+    config::rest_authn_type authn_type{rq.ctx.authn_type};
+    auto handler =
+      [user, authn_type, res_fmt, make_metadata_req, rp = std::move(rp)](
+        kafka_client_cache& cache) mutable -> ss::future<server::reply_t> {
+        client_ptr client{cache.fetch_or_insert(user, authn_type)};
```

Comment:
You may be able to fix the lifetime issue by extending the life of `client_ptr` into a finally block at the end of the continuation.